src/structures/paging/recursive.rs

```diff
@@ -39,17 +42,30 @@ pub trait Mapper<S: PageSize> {
         page: Page<S>,
         frame: PhysFrame<S>,
         flags: PageTableFlags,
-        allocator: A,
+        allocator: &mut A,
     ) -> Result<MapperFlush<S>, MapToError>
     where
-        A: FnMut() -> Option<PhysFrame>;
+        A: FrameAllocator;
+
+    /// Removes a mapping from the page table and returns the frame that used to be mapped.
+    ///
+    /// Note that no page tables or pages are deallocated.
```

Comment:
What do you mean with "or pages"? How about "Note that no page table frames are deallocated."?

Reply:
Meaning pages other than page tables.